modules/websearch/lib/search_engine_obelix.py

```diff
@@ -0,0 +1,561 @@
+        :return: a float, the final score for the recid
+        """
+
+        recommendations_impact = self.settings.recommendations_impact
+        order_score = self.ranked_scores_by_order[recid_index] * (1 - recommendations_impact)
+
+        if recid in self.recommendations:
+            recommended_score = self.recommendations[recid] * recommendations_impact
+            return order_score + recommended_score
+
+        return order_score
+
+    def _get_hitset(self):
+        """
+        Returns the origial hitset list[int,int], for instance: [1,2,3]
+        :return:
+            The original hitset passed to the object, may be an empty list
+        """
+        return self.__hitset
+
+    def _rank_records_by_order(self):
+        """
+        Rank the records by the original order they we're provided
+
+        If the SearchEngineObelix([1,2,3]) then this method will return [1.0, 0.933, 0.867]
+
+        Typically called like this:
+            records, scores = __build_ranked_by_order()
+
+        :return:
+            A tuple, one list with records and one with scores.
+            The list of records are integers while the scores are floats: [1,2,3],[.9,.8,7] etc...
+
+        """
+
+        if len(self._get_hitset()) == 1:
+            return self.__hitset, [self.settings.score_one_result]
+
+        upper = 1
+        lower = self.settings.score_lower_limit
+        size = len(self._get_hitset())
+
+        if size < self.settings.score_min_limit:
+            size *= self.settings.score_min_multiply
+
+        step = ((upper - lower) * 1.0 / size)
+        scores = [1 - (lower + i * step) for i in range(0, size)]
+
+        return self._get_hitset(), scores[0:len(self._get_hitset())]
+
+    def _sort_records_by_score(self, record_scores):
+        """
+        :param record_scores: dictionary {1:0.2, 2:0.3 etc...
+        :return: a tuple with two lists, the first is a list of records and the second of scores
+        """
+        records = []
+        scores = []
+
+        for recid in sorted(record_scores, key=record_scores.get, reverse=True):
+            records.append(recid)
+            scores.append(record_scores[recid])
+
+        return records, scores
+
+
+class ObelixSearchEngineSettings(object):
+    """ Obelix Settings object """
+
+    def __init__(self, redis=None):
+        """
+        Sets the default settings values, may specify redis service to use (mocking)
+        :param redis:
+        :return:
+        """
+
+        if redis:
+            self.redis = redis
+        else:
+            self.redis = obelix_get_redis()
+
+        self.redis_prefix = "obelix::"
+        self.recommendations_impact = 0.5
+        self.score_upper_limit = 1
+        self.score_lower_limit = 0.2
+        self.score_min_limit = 10
+        self.score_min_multiply = 4
+        self.score_one_result = 1.0
+        self.method_switch_limit = 20.0
+        self.redis_timeout_search_result = 300
```

Comment:
It depends on whether this values can be changed or not.

If they can be change  depending on the "installation needs", then it should go as a dictionary on `invenio.conf`.
Otherwise a simple dictionary `DEFAULT_CONFIG_VALUES` on the same file would of the work.